x_xy/io/xml/abstract.py

```diff
@@ -111,49 +111,33 @@ def to_xml(element: T, pos_min: jax.Array, pos_max: jax.Array):
         element.set("pos_max", _to_str(pos_max))
 
 
-def _from_xml_vispy(attr: ATTR):
-    """Find all keys starting with `vispy_`, and return subdict without that prefix.
-    Also convert all arrays back to list[float], because of `struct.field(False)`.
-    Otherwise jitted functions with `sys` input will error on second execution, since
-    it can't compare the two vispy_color arrays.
-    """
+def _to_xml_colors(element: T, geom: base.Geometry) -> None:
+    "Copy pasted from you"
 
-    def delete_prefix(key):
-        len_suffix = len(key.split("_")[0]) + 1
-        return key[len_suffix:]
+    if hasattr(geom, "color") and geom.color is not None:
+        element.set("color", _to_str(geom.color))
 
-    dict_no_prefix = {
-        delete_prefix(k): attr[k] for k in attr if k.split("_")[0] == "vispy"
-    }
+    if hasattr(geom, "edge_color") and geom.edge_color is not None:
+        element.set("edge_color", _to_str(geom.edge_color))
 
-    # convert arrays -> list[float]
-    to_list = (
-        lambda ele: ele.tolist() if isinstance(ele, (np.ndarray, jax.Array)) else ele
-    )
-    return {key: to_list(value) for key, value in dict_no_prefix.items()}
```

Comment:
I believe you will still need this logic, from the xml `color="0.5 0.5 0.5 1" will be interpreted as an array input and automatically lifted to an array. But for fields which have `pytree_node=False` this will cause a bug. Thus, convert back to list[float].

Reply:
updated `_from_xml_geom_attr_processing` to convert jax.Arrays to tuples for color and edge_color